### DISCUSSION TITLE
[Design/General] Economy Overhaul - Space Cash & Supply Points

### DISCUSSION THREAD

**Discussion by Sirryan2002** [ParadiseSS13/Paradise#18652] [Documents]
# Economy Rework - Space Cash & Supply Points
This design began with my foray into refactoring our in-game monetary systems (which are atrocious and haven't been touched in a decade except for TGUI stuff). Little serious thought has actually been put into what numbers we use in our values for resources, and when I speak to the in-game economy I don’t just mean our money/financial systems but also the energy/power economy, the botanical/culinary economy, and supply economy. My hope with this design is to find positive feedback loops and remove or seriously dial back them as well as identify resource taps/sinks and regulate them better than we have in the past. This doc specifically will start only with Space Credits and Supply Department related mechanics, however my plan is to eventually to move to our other economic systems soonish.
## Preface: Current Issues
**Issue 1:** Our Central Currency-Space Cash” or “Space Credits”- is effectively valueless. It fails the three requirements of a currency which is that it is a store of value (it holds relatively the same value over time), it is a unit of account (I can use it like a measuring stick to measure the value of an exchange/item), and that it is a medium of exchange (it’s acceptable to players to receive space credits in exchange for another item). Space Credits fulfill none of those three requirements. The only three key places where space cash holds any value is:

- Sol Traders - sol traders set arbitrary amounts for what their items cost so crew must dig into their accounts to exchange space cash for fun items (that actually have worth)
- Merch Computer - This is the only vendor that prices items at a cost that is reasonable to their worth relative to the monetary base in-game. However, everything here is cosmetic and only holds social value.
- Regular paid vendors barely count as the cost of a single purchase is so insignificantly small that it can be treated as a free item. On average, a crew member can buy up to 90 vendor items with their round-start balance, they usually will not buy more than 5 however (this is a guesstimation but probably accurate). 

Outside of those three uses, money holds no value. So most in-game scenarios cannot use space credits without the requirement of roleplaying/pretending that space cash means something.

---

**Issue 2:** There is an over surplus of Space Cash in-game relative to set item cost. Currently in an average 75 player round, there are roughly $315,000 (in money accounts) and $100,000 (hard cash) space credits in circulation that are possessed by the station or station crew. However, the actual total value of all items sold in vendors only amounts to $150,000 (roughly) on boxstation and is very similar on other stations. The money supply vastly drowns out the actual value of supplied goods (not to be confused with what the crew would actually buy which would be significantly less).  

---

**Issue 3:** At the moment, `resources -> space cash` conversions are completely unbalanced due to an overabundance of resources or bad conversion ratios. This means that space cash will remain unusable in and disconnected from most systems until their values are balanced and reworked:

- Energy power production and consumption atm is completely borked, some atmospheric engines can be pushed so hard to their upper limit that they easily surpass the station’s energy needs by a factor of 100. Additionally, there is no set standard for energy consumption in in-game machinery-this also causes positive feedback loops as some machinery uses less power when upgraded.
- TC can be converted at a ratio of 1TC to 5000 space credits. This means on a generic traitor round there can be upwards of $1 Million space cash worth of assets sitting in traitor uplinks
- Botany and Kitchen Related items can be mass produced which means they cannot be sold for a value atm because it would quickly flood the market with space cash. While a simulated supply/demand function could deal with this, we don’t actually know just how much food/plants can be produced atm given an incentive for players to actually produce as much as possible to generate space cash.
- Xenobiology can mass produce construction materials, food, and other fun items (such as yellow slime cores). This has the same issues as botany/kitchen.
- Chemicals are completely infinite (this could be fixed by balancing the energy system) and therefore cannot be given a monetary value as they have no value except set by the entry cost of possessing a chem dispenser.
- Research levels can be maxed very quickly and sold for an extreme amount of cargo points halfway through the round or earlier

Right now only the mining points system can be connected to Space Cash as it's the only system that has some measure of resource balance. 

---

**Issue 4:** We have little data on our in-game economy. We have good data on cargo crates and job objectives completed; That's it. There’s no way to determine what the actual “demand” is for vendor related items so it’s hard to set a good price. We also can’t tell how much ore is deposited in the ORM and what damage the 2MP -> 1 Space Cash conversion will do to the economy either. 
## Proposed Solutions:
**Reduce the in-game money supply:** The crew's starting balances need to be cut down to 500, whether or not heads get additional funds is not something I’ve figured out yet. The amount of hard cash on stations needs to be cut down to less than 10k or be absorbed into the station round-start money account. The total station budget needs to be cut down to something. I won’t know actual specific numbers for round-start balances until I start trying to connect everything together. I intend on making round-start accounts be pretty low in credits so that we can implement passive income + job objectives from players' occupations in-game.

---

**Increase money demand:** This is a pretty straightforward thing: add more things that cost money, make more free things cost money. Eating from a vendor should take a significant chunk out of your balance, people shouldn’t just be able to empty the public utility vendors for free. We should really be locking better items (such as tool belts and QOL items behind credit costs). 

---

**Make Cargo Crates Cost Credits:** Departments should be getting a budget with a passive income + commissions from job objectives. They should also be forced to use them by making departments pay for their own crates. This will also allow a measure of control over what can be ordered. I’m still not sure whether or not I want to give the captain/HoP the ability to move the station budget around how they please.

---

**Blackbox Data:** Our entire economy at the moment should be tracked with blackbox data:

- How many ores are deposited in the ORM per round?
- How many times are vendors used in a round?
- How often and how much money is being circulated as opposed to sitting in accounts all round
- How much money flows into the station and how much flows out?

---

**Establishment of Resource Standards:** There is nowhere where we actually set a standard for what values are ok. This is why our power system makes no sense and why our money supply far outweighs the money demand. We should be extensively documenting what types of items should cost what.

---

**Remove Problematic Converters:** the 1TC space cash briefcase needs to be cut down to like 1000 space cash or the TC cost upped to like 5TC. 

## Things that we should not be adding to our economy:
People have a lot of ideas and other codebases have plenty of various implementations of their game economies. Here’s a few things I really don’t want to entertain.

---

**Attaching Values to Most Items:** This is a controversial topic, should the station just be able to sell any item? Absolutely not. At the moment, resources are unbalanced. All it takes is for someone to find a way to mass produce a certain item and the economy is toast. Even limiting it to departmental items is dangerous as research, botany, the kitchen, and chemistry can all mass produce lots of different items, so much so that players are occasionally warned for producing items in the thousands… As a side note, this is a nightmare to code. Instead we should consider bounties and job objectives.

---

**Everything should cost money:** While Nanotrasen is a super corporation, not everything in-game should cost money. Certain round-start resources for departments are 100% crucial to allowing that department to progress further into the round or function at all. Furthermore, it makes no sense to make crew members drain 80% of their roundstart account to get the most base level equipment. The economy should influence and direct player/group behavior but not dominate and control it. i.e. Push, don't shove.

---

**Persistent Economy:** Absolutely not, the advantage of the reset button being hit every round is that we don't have to track long term issues with our economy system, just what happens during a single round. I don't want to balance an economy around it persisting through multiple different gamemodes (wizard summon guns *shiver*). Besides there is not much fun in the market becoming unbalanced the next round and either making progression incredibly easy or incredibly slow, as it just upsets a balance we try hard to set. On the coding side, this requires some serious DB magic that I'm not going to invest a month of my life into. In the future there could be persistence of item markets or virtual asset values, but that's very low on my priority list.

---

**(FOR THE MOMENT BUT MAYBE LATER) Creating Crypto, Stock Markets, Supply & Demand:** Economics is complex, you can’t build an entire simulated game market for items and then just expect to hook all of our systems into it. Until we balance our other systems and are confident in the stability of Space Cash, we shouldn’t complicate our WIP space cash system. This is why bounties are a great alternative: they’re simple and we can limit the bounties/rewards offered per round.

## Why?:

Game economies are an extraordinarily powerful tool that we don’t use enough here. The best examples of effective economy in-game are the TC/uplink, changeling points, vampire usable blood, terror spider balance, and… that's it. Those are examples of where we force players to make choices that have significant upsides and downsides that actually have meaning because of resource scarcity. Want to go loud as a traitor? It will cost lots of TC. Want to choose a certain changeling ability? Well you need to out yourself and absorb someone and if you need to use another ability down the line you might be out of luck if you choose now. As a terror queen, do you spend your special on a Bruiser Prince, a Support Mother, or a Resource Producer Princess?

These are the kinds of choices we need to force departments to make. Why does security just get to stock their armory full of every single piece of sec equipment available from the cargo shuttle? Why does research just to upgrade every piece of equipment on a station have absolutely no consequences? Why is that 30 minutes into the round, every item possible to attain in our department progression systems is available to the crew. 

## Where I’m at now:

The economy used to be based in a single computer on station, now it’s in a subsystem and moved into a database datum. The “backend” economy code has be for the most part, completely refactored, I’m just in the process of hooking it fully into stuff like the Arcade Machine, EFTPOS, Vendors, ATMs, etc.

I hope to begin converting Supply Points over to space cash soon so I can begin balancing values.

**Comment by roarbark** [comment]
jobs that require vendor items, say security or medical especially could be given a certain amount of "loadout points" or something the like to spend on their vendors

the idea would be that they wouldn't have to spend all of their starting cash for crucial items- the amount of points could be adjusted depending on if the community wants for example doctors to link up, communicate and strategize on who spends their points on what at first or if a more casual handholdy status quo thing is wanted where you just get your items and click people with it.

**Reply by Inadvizable** [reply]
Or they could use their departmental resources to buy them?

**Reply by Sirryan2002** [reply]
Viz makes a good point about using loadout points as opposed to just having this be departmental space credits. I don't know how I want to handle most round-start loadout stuff, b/c it already costs money to purchase the vendor restocking unit from cargo.
